stdlib/public/runtime/SwiftDtoa.cpp

```diff
@@ -993,6 +1168,60 @@ static size_t swift_format_constant(char *dest, size_t length, const char *s) {
 }
 #endif
 
+#if SWIFT_DTOA_FLOAT16_SUPPORT
+size_t swift_format_float16(const __fp16 *d, char *dest, size_t length)
+{
+    uint16_t raw = *(const uint16_t *)d;
+    if ((raw & 0x7c00) == 0x7c00) { // Infinite or NaN
+        if (raw == 0x7c00) {
+            return swift_format_constant(dest, length, "inf");
+        } else if (raw == 0xfc00) {
+            return swift_format_constant(dest, length, "-inf");
+        } else {
+            // NaN
+            static const int significandBitCount = 10;
+            const char *sign = (raw & 0x8000) ? "-" : "";
+            // XXX FIXME: This is different from Float/Double because
+            // Swift's Float16 NaN encoding gets altered before it gets
+            // here, probably because it gets converted to Float and back.
+            const char *signaling = ((raw >> (significandBitCount - 2)) & 1) ? "s" : "";
```

Comment:
Let's not try to do this. In particular, this will erroneously print some qNaNs as sNaN, which is significantly worse than printing a sNaN as qNaN.

Reply:
I reverted it to use the same detection as Float/Double.  That sacrifices sNaN printing in the short term, but it should DTRT when we get better argument passing for Float16.